tests/src/Functional/DatasetBTBTest.php

```diff
@@ -129,6 +128,68 @@ public function testHarvestOrphan() {
     $this->assertEquals('published', $this->getModerationState('4'));
   }
 
+  /**
+   * Test cleanup of orphaned draft distributions.
+   */
+  public function testOrphanDraftDistributionCleanup() {
+    // Get the original configuration settings.
+    $config = \Drupal::service('config.factory');
+    $datastoreSettings = $config->getEditable('datastore.settings');
+    $deleteLocalResourceOriginal = $datastoreSettings->get('delete_local_resource');
+    $triggeringOriginal = $datastoreSettings->get('triggering_properties');
+    $defaultModerationState = $config->getEditable('workflows.workflow.dkan_publishing');
+    $defaultModerationStateOriginal = $defaultModerationState->get('type_settings.default_moderation_state');
+
+    // Set delete local resource files = false.
+    $datastoreSettings->set('delete_local_resource', 0);
+
+    // Set modified as a triggering property.
+    $datastoreSettings->set('triggering_properties', ['modified'])->save();
+
+    // Set default moderation state = draft.
+    $defaultModerationState->set('type_settings.default_moderation_state', 'draft')->save();
+
+    // Post dataset 1 and run the 'datastore_import' queue.
+    $id_1 = uniqid(__FUNCTION__ . '1');
+    $this->storeDatasetRunQueues($id_1, '1', ['1.csv']);
+
+    // Get the dataset info.
+    $metadata =  \Drupal::service('dkan.common.dataset_info')->gather($id_1);
+    $distributionTable = $metadata['latest_revision']['distributions'][0]['table_name'];
+
+    // Confirm distribution table exists.
+    $databaseSchema = $this->container->get('database')->schema();
+    $distributionTableExists = $databaseSchema->tableExists($distributionTable);
+    $this->assertTrue($distributionTableExists, $distributionTable . ' exists.');
+
+    // Get the associated distribution's resource directory
+    $resourceId = $metadata['latest_revision']['distributions'][0]['resource_id'];
+    $resourceVersion = $metadata['latest_revision']['distributions'][0]['resource_version'];
+    $resourceDirectory = $resourceId . '_' . $resourceVersion;
+
+    // Confirm distribution local directory exists.
+    $this->assertDirectoryExists('public://resources/' . $resourceDirectory);
+
+    // Update the modified date for the dataset.
+    $this->getMetastore()->patch('dataset', $id_1, json_encode(['modified' => '06-05-2020']));
+
+    // Simulate datastore_import and cleanup queues post update.
+    $this->runQueues(['datastore_import', 'orphan_reference_processor']);
+
+    // Confirm original distribution table removed.
+    $distributionTableExists = $databaseSchema->tableExists($distributionTable);
+    $this->assertFalse($distributionTableExists, $distributionTable . ' removed.');
+
+    // Confirm original distribution local directory removed.
+    $this->assertDirectoryDoesNotExist('public://resources/' . $resourceDirectory);
+
+    // Restore the original config values.
```

Comment:
As above, don't need to restore state because BTB will just throw away this drupal instance.